confd/pkg/backends/calico/client.go

```diff
@@ -1367,38 +1387,67 @@ func (c *client) GetLoadBalancerIPs() []*net.IPNet {
 
 // "Global" here means the routes for cluster IP and external IP CIDRs that are advertised from
 // every node in the cluster.
-func (c *client) updateGlobalRoutes(current, new []string) error {
-	for _, n := range new {
+func (c *client) updateGlobalRoutes(routes []string, programmedRejectRoutes, programmedRoutes map[string]bool) error {
+	for _, n := range routes {
 		_, _, err := net.ParseCIDR(n)
 		if err != nil {
 			// Shouldn't ever happen, given prior validation.
 			return err
 		}
 	}
 
-	// Find any currently advertised CIDRs that we should withdraw.
-	withdraws := []string{}
-	for _, existing := range current {
-		if !contains(new, existing) {
-			withdraws = append(withdraws, existing)
-		}
-	}
-
 	if !c.ExcludeServiceAdvertisement() {
-		// Withdraw the old CIDRs and add the new.
 		log.Info("Advertise global service ranges from this node")
-		c.addRoutesLockHeld(rejectKeyPrefix, rejectKeyPrefixV6, new)
-		c.addRoutesLockHeld(routeKeyPrefix, routeKeyPrefixV6, new)
-		c.deleteRoutesLockHeld(rejectKeyPrefix, rejectKeyPrefixV6, withdraws)
-		c.deleteRoutesLockHeld(routeKeyPrefix, routeKeyPrefixV6, withdraws)
+		for _, r := range routes {
```

Comment:
This can all be simplified. It works, but the current form is more verbose than it needs to be ( can combine some branches)